tools/src/test/java/org/apache/kafka/tools/GetOffsetShellTest.java

```diff
@@ -52,20 +53,24 @@ public class GetOffsetShellTest {
     private final int topicCount = 4;
     private final int offsetTopicPartitionCount = 4;
     private final ClusterInstance cluster;
+    private final String topicName = "topic";
 
     public GetOffsetShellTest(ClusterInstance cluster) {
         this.cluster = cluster;
     }
 
     private String getTopicName(int i) {
-        return "topic" + i;
+        return topicName + i;
     }
 
-    public void setUp() {
+    @BeforeEach
+    public void before() {
         cluster.config().serverProperties().put("auto.create.topics.enable", false);
         cluster.config().serverProperties().put("offsets.topic.replication.factor", "1");
         cluster.config().serverProperties().put("offsets.topic.num.partitions", String.valueOf(offsetTopicPartitionCount));
+    }
 
+    public void setUp() {
```

Comment:
Why do we need to split the logic between `before` and `setUp`?

Reply:
The `setUp` is necessary because when in `before` (i.e. `beforeEach`), the kafka cluster is still not created yet. That's why we can inject custom broker properties there. And in `setUp`, we can create producer/admin to talk to the brokers.